### PR TITLE
luaposix: 33.4.0 -> 34.0.4, add dependencies 

### PR DESCRIPTION
**Pull request by lblasc**
###### Motivation for this change

Refresh package and add new dependencies.

Thanks!

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by lblasc** [comment]
@vyp please take a look, thanks!

**Comment by lblasc** [comment]
@Mic92 maybe you can take a look? Thanks!

**Comment by Mic92** [comment]
@GrahamcOfBorg build luajitPackages.bit32 lua52Packages.bit32 lua51Packages.bit32 lua51Packages.luaposix lua52Packages.luaposix luajitPackages.luaposix

**Comment by Mic92** [comment]
It looks like darwin mixes multiple lua versions.

**Comment by Mic92** [comment]
@GrahamcOfBorg build lua52Packages.luaposix

File: pkgs/top-level/lua-packages.nix [hunk 1]
```diff
@@ -43,6 +43,35 @@ let
...
+
+    src = fetchFromGitHub {
+      owner = "keplerproject";
+      repo = "lua-compat-5.2";
+      rev = "bitlib-${version}";
+      sha256 = "1ipqlbvb5w394qwhm2f3w6pdrgy8v4q8sps5hh3pqz14dcqwakhj";
+    };
+
+    buildPhase = ''
+      cc -shared -I${lua}/include -Ic-api lbitlib.c -o bit32.so
```
Comment (Mic92) [inline pkgs/top-level/lua-packages.nix]: Can you try to add `lua` to buildInputs instead?

**Comment by Mic92** [comment]
@GrahamcOfBorg build lua52Packages.bit32

**Comment by Mic92** [comment]
@GrahamcOfBorg build lua51Packages.bit32

**Comment by Mic92** [comment]
@GrahamcOfBorg build lua52Packages.bit32

**Comment by Mic92** [comment]
@GrahamcOfBorg build lua52Packages.luaposix

**Comment by Mic92** [comment]
@GrahamcOfBorg build luajitPackages.bit32 lua52Packages.bit32 lua51Packages.bit32 lua51Packages.luaposix lua52Packages.luaposix luajitPackages.luaposix

File: pkgs/top-level/lua-packages.nix [hunk 2]
```diff
@@ -270,22 +299,38 @@ let
...
+    propagatedBuildInputs = [ std.normalize bit32 ];
+
+    buildPhase = ''
+      ${lua}/bin/lua build-aux/luke \
+        package="luaposix" \
+        version="${version}" \
+        LUA="${lua}/bin/lua" \
+        LUA_INCDIR="-I${lua}/include" \
+        INST_LIBDIR="$out/lib/lua/${lua.luaversion}" \
+        INST_LUADIR="$out/share/lua/${lua.luaversion}"
```
Comment (Mic92) [inline pkgs/top-level/lua-packages.nix]: I was able to remove the last 4 lines here. Are they really necessary?

File: pkgs/top-level/lua-packages.nix [hunk 3]
```diff
@@ -270,22 +299,38 @@ let
...
     src = fetchFromGitHub {
       owner = "luaposix";
       repo = "luaposix";
       rev = "release-v${version}";
-      sha256 = "0y531p54lx2yf243bcsyp6sv8fvbqidp20yry0xvb85p8zw9dlrq";
+      sha256 = "0p5583vidsm7s97zihf47c34vscwgbl86axrnj44j328v45kxb2z";
     };
 
-    buildInputs = [ perl ];
+    propagatedBuildInputs = [ std.normalize bit32 ];
```
Comment (Mic92) [inline pkgs/top-level/lua-packages.nix]: Is `bit32` needed for lua5.2?

Comment (lblasc) [reply pkgs/top-level/lua-packages.nix]: Seems like luaposix library doesn't check if running under lua5.2, it loads bit32 lib every time. I would leave it.

Comment (Mic92) [reply pkgs/top-level/lua-packages.nix]: ok.

**Review by lblasc — COMMENTED** [review]
removed those unneeded build options, it works.
Thanks for the review!

File: pkgs/top-level/lua-packages.nix (hunk 2 above)
Comment (Mic92) [reply pkgs/top-level/lua-packages.nix]: mhm. Version should be there: https://github.com/luaposix/luaposix/blob/649454019dcc983eb6b398624e5b6c68d72f7808/lib/posix/version.lua.in

Comment (Mic92) [reply pkgs/top-level/lua-packages.nix]: nevermind it is still there.

**Comment by Mic92** [comment]
I rebased everything in 13c8e0194d6f3d948332e564a68b275834152c25

**Pull request closed without merging**